[yitong241] iP

yitong241 opened the pull request:
# DUKE
> "Just use DUKE"      -yitong

DUKE is:
- ~~easy~~ SUPER easy to learn 😎
- easy to use

**Download** it [here](https://github.com/yitong241/ip).

Supported *operations*:
1. add
2. delete
3. list
4. mark as done
5. mark as undone

You can run the GUI with `Launcher`:
```java
public class Launcher {
    public static void main(String[] args) {
        Application.launch(Main.class, args);
    }
}
```

Features:
- [x] todo
- [x] deadline
- [x] event

Commit:
Add Gradle support

yitong241 committed:
Initial build of Duke

yitong241 committed:
Add some java doc

yitong241 committed:
Add all java doc

yitong241 committed:
Remove unmappable character from ASCII

yitong241 committed:
bug fix

yitong241 committed:
bug fix

yitong241 committed:
bug fix

yitong241 committed:
bug fix

yitong241 committed:
bug fix

yitong241 committed:
Merge remote-tracking branch 'origin/master'

# Conflicts:
#	src/main/java/Tasks/Monitor.java
#	text-ui-test/EXPECTED.TXT

yitong241 committed:
allow tasks to be stored in local disk

yitong241 committed:
add LocalDateTime

yitong241 committed:
modification for event

yitong241 committed:
add support for date and time

yitong241 committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	src/main/java/Tasks/Event.java

yitong241 committed:
merge master and branch-Level-8

yitong241 committed:
put all classes in one package

yitong241 committed:
Merge branch 'add-gradle-support'

yitong241 committed:
add JUnit Test

yitong241 committed:
add JUnit Test

yitong241 committed:
add more javaDoc

yitong241 committed:
add find method

yitong241 committed:
add find method

yitong241 committed:
add find method

yitong241 committed:
add more java doc

yitong241 committed:
add more java doc

yitong241 committed:
Tweak the code to comply with a coding standard

yitong241 committed:
Tweak the code to comply with a coding standard

yitong241 committed:
add find.java

yitong241 committed:
add find.java

yitong241 committed:
bug fix

yitong241 committed:
add find

yitong241 committed:
Merge branch 'branch-A-JavaDoc'

yitong241 committed:
Merge branch 'branch-A-CodingStandard'

yitong241 committed:
Merge branch 'branch-Level-9'

# Conflicts:
#	src/main/java/Duke/Interpreter.java

yitong241 committed:
Check the style for all java files

NoyaRoeT commented on src/main/java/Duke/Interpreter.java:
```diff
@@ -0,0 +1,247 @@
+     */
+    public static Task mark(String command, TaskTable table) throws InvalidCommandException, OutRangeException {
+        int len = table.size();
+        int index;
```
The Java coding standard says to initialize variables where they are declared. Would it be better to declare index at line 39 instead?

NoyaRoeT commented on src/main/java/Duke/Tasks/Disk.java:
```diff
@@ -0,0 +1,75 @@
+
+public class Disk {
+
+    private final String path;
+    private final DateTimeFormatter format = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
```
Perhaps using SCREAMING_SNAKE_CASE for constant names would better match the Java coding standards?

NoyaRoeT commented on src/main/java/Duke/Tasks/Deadline.java:
```diff
@@ -0,0 +1,78 @@
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents the Deadline class
```
Would it be redundant to say that the Deadline class represents the Deadline class? I noticed the same thing in your other classes as well.

NoyaRoeT commented on src/main/java/Duke/Tasks/Event.java:
```diff
@@ -0,0 +1,88 @@
+
+    /**
+     * The constructor of Event
+     * @param desc the description of event
+     * @param startTime the start time of event
+     * @param endTime the end time of event
+     * @param done whether the event is done
```
A very small thing, but Java standards suggest adding punctuation behind each parameter description.

NoyaRoeT reviewed (commented):
Overall, I found your code easy to read for the most part. I noted a few minor coding standard violations and some of the header comments also seem redundant.

yitong241 committed:
add functions for GUI

yitong241 committed:
merge with branch-level-10

yitong241 committed:
bug fix of GUI

yitong241 committed:
add instance of Varargs

yitong241 committed:
Interpreter class: add assertions

Index related functions (e.g. mark(), unmark(), delete() etc.) need to ensure the input index are within valid range.

The out-ranged input index causes unexpected errors.

Adding the assertions into the index related functions allows the program to check the validity of the input. For example, when input index is smaller than 0, AssertionError will be raised.

Let’s add the assertions at necessary places to reduce the unexpected errors raised.

Refer to this tutorial on dealing with assertions
https://www.baeldung.com/java-assert

yitong241 committed:
Interpreter class: remove dead code

Interpreter class have some commented dead code which could never be reached.

The out-dated dead code decreases the readability of the code.

Deleting the dead code could follow the Code Quality better. This also allows the error not to be caught at the interpreter class, instead, it will be passed to execute() function Duke.java.

Let’s not handle the errors raised in the interprete() stage, instead, we handle it in the execute() function in Duke.java. This allows the errorMessage to be passed to the GUI and displayed to the users.

Refer to this tutorial on Code Quality
https://nus-cs2103-ay2223s2.github.io/website/se-book-adapted/chapters/codeQuality.html#code-quality

yitong241 committed:
Add Stream

yitong241 committed:
Merge pull request #1 from yitong241/branch-A-Assertions

Interpreter class: add assertions

yitong241 committed:
Merge branch 'branch-A-CodeQuality'

yitong241 committed:
Merge with branch-A-assertions

yitong241 committed:
Merge branch 'branch-A-Streams'

yitong241 committed:
Set up CI

yitong241 committed:
Add the ability to recognize and deal with duplicate items.
e.g. tasks with same name

yitong241 committed:
Bug fix for runtest.sh

yitong241 committed:
Bug fix for runtest.sh

yitong241 committed:
Bug fix

yitong241 committed:
Bug fix

yitong241 committed:
Delete README.md

yitong241 committed:
add user guide

yitong241 committed:
Merge remote-tracking branch 'origin/master'

yitong241 committed:
bug fix

yitong241 committed:
bug fix
